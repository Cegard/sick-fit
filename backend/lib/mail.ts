import { createTransport } from 'nodemailer';

const transport = createTransport({
    host: process.env.MAIL_HOST,
    port: process.env.MAIL_PORT,
    auth: { user: process.env.MAIL_USER, pass: process.env.MAIL_PASS },
});

interface MailResponse {
    message: string;
}

function makeANiceEmail(text: string): string {
    return `
    <div style="
        border: 1px solid black;
        padding: 20px;
        font-family: sans-serif;
        line-height: 2;
        font-size: 20px;
    ">
        <h2> Hey! </h2>
        <p> ${text} </p>
        <p> Bye! </p>
    </div>
    `;
}

export async function sendPasswordResetFunction(
    token: string,
    to: string
): Promise<void> {
    const info = (await transport.sendMail({
        to,
        from: 'test@test.com',
        subject: 'Reset your password',
        html: makeANiceEmail(`
            Click
            <a href="${process.env.FRONTEND_URL}/reset?token=${token}"> here </a>
            to reset your password.
        `),
    })) as MailResponse;

    console.log(info);
}
