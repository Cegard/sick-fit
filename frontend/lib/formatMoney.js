export default function formatMoney(amount = 0) {
  const options = {
    style: 'currency',
    currency: 'USD',
    minimumFranctionDigits: 2,
  };

  if (amount % 100 === 0) {
    options.minimumFranctionDigits = 0;
  }

  const formatter = Intl.NumberFormat('en-US', options);

  return formatter.format(amount / 100);
}
