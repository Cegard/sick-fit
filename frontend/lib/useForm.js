import { useEffect, useState } from 'react';

export default function useForm(initial = {}) {
  const [inputs, setInputs] = useState(initial);
  const initialValues = Object.values(initial).join(' ');

  useEffect(() => {
    setInputs(initial);
  }, [initialValues]);

  function handleChange(e) {
    let { value, type, name } = e.target;

    if (type === 'number') value = value ? parseInt(value) : 0;

    if (type === 'file') [value] = e.target.files;

    setInputs({
      ...inputs,
      [name]: value,
    });
  }

  function resetForm() {
    setInputs(initial);
  }

  function clearForm() {
    const blankState = Object.fromEntries(
      Object.entries(inputs).map(([key, value]) => [key, ''])
    );
    setInputs(blankState);
  }

  return {
    inputs,
    handleChange,
    resetForm,
    clearForm,
  };
}
