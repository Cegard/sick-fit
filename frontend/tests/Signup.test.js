import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import userEvent from '@testing-library/user-event';
import Signup, { SIGNUP_MUTATION } from '../components/SignUp';
import { fakeUser } from '../lib/testUtils';

const me = fakeUser();
const password = 'pass123';

const mocks = [
  {
    request: {
      query: SIGNUP_MUTATION,
      variables: {
        name: me.name,
        email: me.email,
        password,
      },
    },
    result: {
      data: {
        createUser: {
          __typename: 'User',
          id: 'id1',
          email: me.email,
          name: me.name,
        },
      },
    },
  },
];

describe('<Signup />', () => {
  it('Renders and matches snapshot', () => {
    const { container } = render(
      <MockedProvider>
        <Signup />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('Calls the mutation properly', async () => {
    const { container } = render(
      <MockedProvider mocks={mocks}>
        <Signup />
      </MockedProvider>
    );

    const nameInput = screen.getByPlaceholderText('Your name');
    const emailInput = screen.getByPlaceholderText('email@example.com');
    const passwordInput = screen.getByPlaceholderText('Your password');
    const submitButton = screen.getByText('Create Account');

    await userEvent.type(nameInput, me.name);
    await userEvent.type(emailInput, me.email);
    await userEvent.type(passwordInput, password);

    await userEvent.click(submitButton);
    await screen.findByText(`Signed up with ${me.email}`);
  });
});
