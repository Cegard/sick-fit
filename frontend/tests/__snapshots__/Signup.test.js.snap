// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> Renders and matches snapshot 1`] = `
<div>
  <form
    class="Form-sc-1xszr8q-0 iaxOhV"
    method="POST"
  >
    <h2>
       Create an user account 
    </h2>
    <fieldset>
      <label
        for="name"
      >
        Name
        <input
          autocomplete="name"
          name="name"
          placeholder="Your name"
          type="text"
          value=""
        />
      </label>
      <label
        for="email"
      >
        Email
        <input
          autocomplete="email"
          name="email"
          placeholder="email@example.com"
          type="email"
          value=""
        />
      </label>
      <label
        for="password"
      >
        Password
        <input
          name="password"
          placeholder="Your password"
          type="password"
          value=""
        />
      </label>
      <button
        type="submit"
      >
         Create Account 
      </button>
    </fieldset>
  </form>
</div>
`;
